pkg/collector/json_path_collector.go

```diff
@@ -131,3 +143,80 @@ func getPodMetrics(pod *corev1.Pod, scheme, path string, port int) ([]byte, erro
 
 	return data, nil
 }
+
+// intsToFloat64s will convert a slice of int to a slice of float64
+func intsToFloat64s(in []int) (out []float64) {
+	for i, v := range in {
+		out[i] = float64(v)
+	}
+	return
+}
+
+// float32sToFloat64s will convert a slice of float32 to a slice of float64
+func float32sToFloat64s(in []float32) (out []float64) {
+	for i, v := range in {
+		out[i] = float64(v)
+	}
+	return
+}
+
+// reduce will reduce a slice of numbers given a reducer function's name. If it's empty or not recognized, avg is used.
+func reduce(values []float64, reducerFunc string) float64 {
+	switch reducerFunc {
+	case "avg":
+		return avg(values)
+	case "min":
+		return min(values)
+	case "max":
+		return max(values)
+	case "sum":
+		return sum(values)
+	default:
+		return avg(values)
```

Comment:
I would rather this function return an error than use average. Currently misconfigurations would be silently handled.

Reply:
Pushed a change to implement this!